R/updateNullModCond.R

```diff
@@ -1,37 +1,40 @@
 
-### fit a conditional model based on a previously fit model. Default does not re-estimate variance components, but it may. 
+### fit a conditional model based on a previously fit model. Default does not re-estimate variance components, but it may.
 updateNullModCond <- function(nullmod, G, covMatList = NULL,  AIREML.tol = 1e-6, max.iter = 100, drop.zeros = TRUE, verbose = TRUE){
-    
-    ## a few checks that may be transfered to wrapper function: 
+
+    ## a few checks that may be transfered to wrapper function:
     if (nullmod$hetResid & is.null(nullmod$group.idx)) stop("group indices are required for updating the null model")
 
     #if (updateVarComp){ ## this check may be pulled out for wrapper function.
     if (is.null(covMatList)) stop("covMatList is needed for udpating variance components")
     if (nullmod$hetResid & is.null(nullmod$group.idx)) stop("group indices are required for updating variance components")
     #}
 
-    
+    if (is.null(colnames(G))) colnames(G) <- paste0("var_", 1:ncol(G))
     X = cbind(nullmod$model.matrix, G)
-    
+
     if (!nullmod$family$mixedmodel){ ## if it is not a mixed model, re-fit the model. (This includes heterogeneous residuals).
-        
-        return(.fitNullModel(nullmod$outcome, X, covMatList = NULL, 
-                             group.idx = nullmod$group.idx, family = nullmod$family$family))  	
+
+        return(.fitNullModel(nullmod$fit$outcome, X, covMatList = NULL,
+                             group.idx = nullmod$group.idx, family = nullmod$family$family))
     }
-    
-    
+
+
     ### if the function reached this far, nullmod is a mixed model!
     ### Re-fit the model with the new design matrix and start point the varComp
-    ## from the provided nullmod object. 
-    
-    new.nullmod <- .fitNullModel(nullmod$outcome, X, covMatList = covMatList, 
+    ## from the provided nullmod object.
+
+    new.nullmod <- .fitNullModel(nullmod$fit$outcome, X, covMatList = covMatList,
                                  group.idx = nullmod$group.idx, family = nullmod$family$family, start = nullmod$varComp,
                                  AIREML.tol = AIREML.tol, max.iter= max.iter, drop.zeros = drop.zeros, verbose = verbose)
-    
+
     ## add any extra slots
     extra <- setdiff(names(nullmod), names(new.nullmod))
     new.nullmod <- c(new.nullmod, nullmod[extra])
-    return(new.nullmod)  
+    # Update the model string.
+    previous_covar_string <- .modelCovarString(new.nullmod$covars)
+    new.nullmod$covars <- c(new.nullmod$covars, colnames(G))
+    new_covar_string <- .modelCovarString(new.nullmod$covars)
+    new.nullmod$model <- sub(previous_covar_string, new_covar_string, new.nullmod$model)
```

Comment:
The model string is added in the `fitNullModel` wrapper, but this function only calls the internal `.fitNullModel` function so it has to jump through some hoops to update the model string.

Reply:
This function is not currently exported, and is awaiting a wrapper of its own. Like `updateNullModelOutcome`, which is called by `nullModelInvNorm`, it was intended to be an internal function. So would it make sense to handle this part in a future wrapper?

Reply:
Yes, I agree that handling it in a wrapper is better. I could add a comment to this part saying that these lines should be moved to a wrapper once it's written. If we remove them entirely, then whoever wrote the wrapper would have to figure out how to code it again. Thoughts?

Reply:
Leaving it in with a comment sounds like a good plan.

Reply:
I added a comment above these lines.